### PR TITLE
Adding speed and charge alerts.

### PR DESCRIPTION
**Pull request by ebabel**
screen shot 2017-12-13 at 2 37 57 am

### CODE REVIEW

File: app/src/main/java/net/kwatts/powtools/util/SharedPreferencesUtil.java [hunk 1]
```diff
@@ -101,4 +120,41 @@ public void saveMacAddress(String macAdress, String macAddressName) {
     public boolean isMetric() {
         return mSharedPref.getBoolean(METRIC_UNITS, false);
     }
+
+    public float getSpeedAlert() {
+
+        boolean isMetric = App.INSTANCE.getSharedPreferences().isMetric();
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/util/SharedPreferencesUtil.java]: `isMetric()` is a method of this class, no need to use App instance.

Could inline the `isMetric()` call as well.

File: app/src/main/java/net/kwatts/powtools/util/SharedPreferencesUtil.java [hunk 2]
```diff
@@ -101,4 +120,41 @@ public void saveMacAddress(String macAdress, String macAddressName) {
...
+        return mSharedPref.getFloat(SPEED_ALERT, defaultSpeedAlert);
+    }
+
+    public void saveSpeedAlert(float speedAlert) {
+        SharedPreferences.Editor editor = mSharedPref.edit();
+        editor.putFloat(SPEED_ALERT, speedAlert);
+        editor.commit();
+    }
+    public int getChargeAlert() {
+        return mSharedPref.getInt(CHARGE_ALERT, 95);
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/util/SharedPreferencesUtil.java]: nit: Hungarian notation

File: app/src/main/java/net/kwatts/powtools/view/AlertsPresenter.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        try {
+            speed = Float.parseFloat(speedString);
+        } catch (NumberFormatException e) {
+            Timber.e(e);
+            return;
+        }
+        float speedAlert = sharedPreferences.getSpeedAlert();
+        boolean isEnabled = sharedPreferences.getSpeedAlertEnabled();
+
+        if (speed >= speedAlert && !isAlarmPlaying && isEnabled) {
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/view/AlertsPresenter.java]: nit: checking simple `boolean`s before numeric comparison could be slightly more efficient when not enabled. Plus reading an enabled check first is easier to follow.

**Comment by kwatkins** [comment]
Nice, currently working on getting the charging state and cleaning up the BLE stuff, data binding etc is a mess.

**Comment by ebabel** [comment]
> cleaning up the BLE stuff
Make sure to see the branch [bluetooth-mock](https://github.com/ponewheel/android-ponewheel/tree/bluetooth-mock).

**Pull request merged**